lib/setup/setupUpgrade.js

```diff
@@ -67,24 +67,38 @@ function Upgrade(options) {
         let adapters;
         if (dependencies) {
             // if like [{"js-controller": ">=0.10.1"}]
-            if (dependencies instanceof Array) {
+            if (Array.isArray(dependencies)) {
                 adapters = {};
                 dependencies.filter(a => typeof dependencies[a] !== 'string')
```

Comment:
why you filter out all "string" entroies They would be not checked at all with this ... Or do I see something wrong?

Reply:
bf legacy but not applied, because it only returns the filtered array lol ;-) will change